internal/storage/payload_reader.go

```diff
@@ -96,6 +97,23 @@ func (r *PayloadReader) GetBoolFromPayload() ([]bool, error) {
 	return ret, nil
 }
 
+// GetBoolFromPayload2 returns bool slice from payload.
+func (r *PayloadReader) GetBoolFromPayload2() ([]bool, error) {
+	if r.colType != schemapb.DataType_Bool {
+		return nil, fmt.Errorf("failed to get bool from datatype %v", r.colType.String())
+	}
+	reader := r.reader.RowGroup(0).Column(0).(*file.BooleanColumnChunkReader)
```

Comment:
better to check if the conversion ok, this will avoid panic~

Reply:
fixed